smalltalksrc/VMMaker/CogSimStackEntry.class.st

```diff
@@ -134,6 +141,21 @@ CogSimStackEntry >> constant: anObject [
 	^constant := anObject
 ]
 
+{ #category : #'compile abstract instructions' }
+CogSimStackEntry >> copyToReg: reg [
+
+	"Copy the value of the current entry to the register argument.
+	Do not touch the stack"
+	
+	type caseOf: {
+		[SSBaseOffset]	-> [cogit MoveMw: offset r: registerr R: reg].
+		[SSSpill]		-> [cogit MoveMw: offset r: registerr R: reg].
+		[SSConstant]	-> [cogit genMoveConstant: constant R: reg].
+		[SSRegister]	-> [reg ~= registerr
+								ifTrue: [cogit MoveR: registerr R: reg]
+								ifFalse: [cogit Label]] }
```

Comment:
Why a label is created if it's the same register? 
Can we change the condition to use `==` instead?

Reply:
> Why a label is created if it's the same register?

good question. I don't know. It was like that I just renamed this method.

> Can we change the condition to use `==` instead?

Make a suggestion and I accept it ;)

Reply:
I didn't have the eggs to make the suggestion :P

Reply:
```suggestion
		[SSRegister]	-> [reg = registerr
								ifFalse: [cogit MoveR: registerr R: reg]
								ifTrue: [cogit Label]] }
```